packages/authorization-agent/src/authorization-agent.ts

```diff
@@ -5,16 +5,55 @@ import {
   CRUDSocialAgentRegistration,
   CRUDApplicationRegistration,
   ImmutableDataGrant,
-  ReadableWebIdProfile
+  ReadableWebIdProfile,
+  ReadableDataAuthorization,
+  ReadableDataRegistration
 } from '@janeirodigital/interop-data-model';
-import { WhatwgFetch, RdfFetch, fetchWrapper } from '@janeirodigital/interop-utils';
-import { AccessAuthorizationStructure, generateAuthorization } from './authorization';
+import { INTEROP } from '@janeirodigital/interop-namespaces';
+import { WhatwgFetch, RdfFetch, fetchWrapper, iterable2array } from '@janeirodigital/interop-utils';
+import {
+  AccessAuthorizationStructure,
+  generateAuthorization,
+  GrantedAuthorization,
+  NestedDataAuthorizationData
+} from './authorization';
 
 interface AuthorizationAgentDependencies {
   fetch: WhatwgFetch;
   randomUUID(): string;
 }
 
+export interface AgentWithAccess {
+  agent: string;
+  dataAuthorization: string;
+  accessMode: string[];
+}
+
+// TODO: duplicats ShareAuthorization from api-messages (sai-impl-service)
+export type ShareDataInstanceStructure = {
+  applicationId: string;
+  resource: string;
+  accessMode: string[];
+  children: {
+    shapeTree: string;
+    accessMode: string[];
+  }[];
+  agents: string[];
+};
```

Comment:
If these types are not used outside of this class then there's no need to export them.

Reply:
One is used by sai-impl-service and the other one we use in the test